### PR TITLE
Dedupe worker attributes

### PR DESCRIPTION
**Pull request by jerome-benoit**
Signed-off-by: Jérôme Benoit <jerome.benoit@sap.com>

## PR Checklist

- [x] Please add a description of your changes.
- [ ] We need your changes to come with unit tests in order to keep this project in quality and easy to maintain.
- [ ] If your changes contain any new feature please be sure to document it.
- [ ] Please add a link to the open issue or task that this pull request will resolve.
- [x] Add a note to the changelog to indicate the change.

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

If yes please indicate it and migration info into the CHANGELOG.md file.

---

### CODE REVIEW

File: tests/worker/abstract-worker.test.js [hunk 1]
```diff
@@ -15,11 +15,11 @@ describe('Abstract worker test suite', () => {
...
 
-  it('Verify worker default values', () => {
+  it('Verify worker options default values', () => {
     const worker = new ThreadWorker(() => {})
-    expect(worker.maxInactiveTime).toBe(1000 * 60)
-    expect(worker.killBehavior).toBe(KillBehaviors.SOFT)
-    expect(worker.async).toBe(false)
+    expect(worker.opts.maxInactiveTime).toBe(1000 * 60)
+    expect(worker.opts.killBehavior).toBe(KillBehaviors.SOFT)
+    expect(worker.opts.async).toBe(false)
```
Comment (Shinigami92) [inline tests/worker/abstract-worker.test.js:22]: **question:** So this is a breaking change to the public / community that uses our package, isn't it? Is it necessary to introduce this change?

Comment (jerome-benoit) [reply tests/worker/abstract-worker.test.js:22]: The attributes are not public. 
If there's users relying on it, it's their responsibility to not use protected attributes of an abstract class and not yours.
I can document the change but nobody needs to use that abstract class.

Comment (Shinigami92) [reply tests/worker/abstract-worker.test.js:22]: Oh so they were accessed here, cause JS didn't stopped us...
We should write tests also in TypeScript :/

**Review by Shinigami92 — DISMISSED** [review]

File: tests/worker/abstract-worker.test.js (hunk 1 above)
Comment (jerome-benoit) [reply tests/worker/abstract-worker.test.js:22]: Nevertheless, I will document the change for a 2.1.x release, the AbstractWorker class is exported.

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**